src/Microsoft.Azure.WebJobs.Host/Queues/Bindings/QueueBindingProvider.cs

```diff
@@ -187,7 +187,7 @@ private IAsyncCollector<IStorageQueueMessage> BuildFromQueueAttribute(QueueAttri
 
             internal IStorageQueue GetQueue(QueueAttribute attrResolved)
             {
-                var account = Task.Run(() => this._accountProvider.GetStorageAccountAsync(attrResolved, CancellationToken.None)).GetAwaiter().GetResult();
```

Comment:
+1. I think the root of the problem here is that there's some wacky layering problems around when things get validated. 

---
In reply to: [123315937](https://github.com/Azure/azure-webjobs-sdk/pull/1190#discussion_r123315937) [](ancestors = 123315937)